July 2025 What's new blog post (#1711)
withastro/astro.build: src/content/blog/whats-new-july-2025.mdx

- [Integrating the UnJS ipx Image Proxy as an Astro Middleware](https://flori.dev/reads/astro-ipx-middleware)
- [How to Protect Auth Forms with Cloudflare, Supabase SSR, and Astro Actions](https://blog.hijabicoder.dev/how-to-protect-auth-forms-with-cloudflare-supabase-ssr-and-astro-actions)
- [How to install and set up HextaUI in your Astro project](https://www.hextaui.com/docs/ui/installation/astro)
- [Building faster content-driven sites with Astro](https://strapi.io/blog/building-faster-content-driven-sites-with-astro)
- [Composing micro-frontends with Astro’s Server Islands](https://medium.com/@sergio.a.soria/composing-micro-frontends-with-astros-server-islands-88a02728436c)
- [Build a fully customizable bun server for Astro](https://dev.to/elite174/fully-customizable-bun-server-for-astro-2g61)
- [Modify slot children in Astro](https://cassidysmith.dev/posts/modify-astro-slot-children)
- [How to create an interactive testimonial with Astrojs, Tailwind CSS and JavaScript](https://lexingtonthemes.com/tutorials/how-to-create-an-interactive-testimonial-with-astro-tailwind-css-and-javascript/)

> - [How to create a horizontal progress indicator for your Starlight site](https://blog.trueberryless.org/blog/starlight-progress-indicator/)

Reviewer: I really appreciate mentioning this blog twice in Astro blogs, but that's really not necessary 😅
But as I would also not like to lose my streak, feel free to link to Astro related stuff I posted this month 😊

```suggestion
- [How to create a Rehype plugin that turns GitHub links into beautiful badges](https://blog.trueberryless.org/blog/rehype-github-badge-links/)
```

If this doesn't fit, there would also be the awesome "how to write blog posts" trilogy from Lou, Jumper and meself or my most recent story: A very serious blog about earworms! 😂

## Content

Here's what caught our attention this month: 

- 💑 Alex Russel noticed that 11ty and Astro have won the hearts of frontend's best minds when creating a collection of [links and resources for folks getting into performance and platform-oriented web development](https://infrequently.org/2025/07/links/).
- 💳 Read why [Duende is supporting Astro (and Starlight)](https://duendesoftware.com/blog/20250703-astro-open-source-sponsorship) with their open source sponsorship.
- 🚀 Learn why Astro is the [modern take on static site generation](https://dev.to/blamsa0mine/building-my-first-blog-with-astro-5-a-modern-take-on-static-site-generation-1bh2) and should be at the top of your consideration list according to A0mineTV.
- 📈 Discover why [Astro is especially recommended for businesses facing challenges in performance, SEO, and streamlining content management](https://coosy.co.jp/blog/astro-js/) by Coosy. (Japanese)

> - 💄 View the (Astro changelog as a pretty website)[https://astro-changelog.netlify.app/] thanks to Felix Schneider.

Reviewer 2: ```suggestion
- 💄 View the [Astro changelog as a pretty website](https://astro-changelog.netlify.app/) thanks to Felix Schneider.
```

Link fix

Reviewer: omg, I got mentioned here too, I didn't even notice, thank youuuu!!!

Author: I will never not keep getting this wrong sometimes! 😄

  ]}
/> 

### Learn Astro

New to Astro? See the latest introductory videos around the web!

<YouTubeGrid
  videos={[
    { id: "https://www.youtube.com/watch?v=2PWm58usdg0", title: "Astro Beginner Bites - How to Display Markdown Content"},
    { id: "https://www.youtube.com/watch?v=WRCXGIxb72w", title: "Ngulik Astro JS : 3.Frontmatter Script & Styling"},
    { id: "https://www.youtube.com/watch?v=WzM-2DeyClo", title: "Astro Course - Class 02: Layouts, Props, SEO and Tailwind"},
    { id: "https://www.youtube.com/watch?v=d2cI-ZheAzw", title: "Astro Beginner Bites - What is Content Collections?"},
    { id: "https://www.youtube.com/watch?v=GsM7MASlzAo", title: "Create a Slick Countries App with Astro and Static Site Generation"},
  ]}
/> 

### Meetups

Astro is popular around the globe! Discover what happened at meetups and conferences this month.

<YouTubeGrid
  videos={[
    { id: "https://www.youtube.com/watch?v=a54JRtqMXfs", title: "techNL Astro Workshop 2025 - Jack Harrhy"},
    { id: "https://www.youtube.com/watch?v=GPaAGRM52L4", title: "Astro: Lightning-fast content-driven websites by Volodymyr Baranchuk"},

>     { id: "https://www.youtube.com/watch?v=zXWufnkl0y0", title: "Le pedí a ChatGPT que hiciera mi portafolio con Astro y Tailwind… y esto pasó"},

Reviewer 3: This one (https://www.youtube.com/watch?v=zXWufnkl0y0) doesn't look like a Meetup! And it is already showcased in "Live coding streams". Wrong link?

- [Astro ETag](https://github.com/Ovyerus/astro-etag) - An Astro integration for generating ETag sidecar files for your build.
- [Astro Normalize Trailing Slash](https://github.com/ReunMedia/astro-normalize-trailing-slash) - A simple utility to normalize trailing slashes in URLs to match your Astro project's `trailingSlash`.
- [Astro Posthog](https://github.com/jop-software/astro-posthog) - An Astro Integration for PostHog, GDRP friendly tracking in your astro app / site.
- [Astro Obfuscator](https://github.com/CarlosSLoureiro/astro-obfuscator) - An Astro integration that allows you to obfuscates JavaScript client files using `javascript-obfuscator`.
- [Astro Get Remote Image](https://github.com/WALKAL0NE/astro-get-remote-img) - An Astro integration that automatically downloads remote images from your CMS during the build process and updates HTML references to use local paths.
- [Astro Client Pragma](https://github.com/mohamedradime/astro-client-pragma) - An Astro integration that auto-adds the proper `client:* directive` to every template by reading a `"client:*";` pragma at the top of each framework component.

> - [Astro Web Vitals](https://github.com/withstudiocms/web-vitals) - An Astro integration that enables tracking real-world website performance and storing the data in Astro DB.

Reviewer 2: ```suggestion
- [Astro Web Vitals](https://github.com/withstudiocms/web-vitals) - An Astro integration that enables tracking real-world website performance and storing the data in Astro DB.
- 🌟 [`starlight-auto-drafts`](https://github.com/HiDeoo/starlight-auto-drafts) - Starlight plugin to tweak draft pages default behavior and automatically remove sidebar links to draft pages in production mode
```

I totally forgot I released a new plugin this month :sweat_smile:

Author: Excellent, lol! Just tweaked the name since we've discovered that our styling for linked inline code is terrible, so we just go to "names" for these!

- [Dev Logs](https://github.com/23Aryansharma11/logs-template) - Start logging your day with a minimal template. 

## Showcase

Looking for even more inspiration? See all the community member websites submitted to [the Astro showcase](https://astro.build/showcase/) this month. [Submit your site built with Astro!](https://astro.build/showcase/submit/)

{/* 
<MonthlyShowcaseGrid startDate={frontmatter.startDate} endDate={frontmatter.endDate} /> 
*/}

export const extraSitesToExclude = [
  'spellcast.fm',
];

export const excludedSites = [...featuredSites, ...monthlySites, ...canvasSites, ...extraSitesToExclude]

<MonthlyShowcaseGrid startDate={frontmatter.startDate} endDate={frontmatter.endDate} exclude={excludedSites} /> 

## Starlight in the wild

It's always exciting to stumble across more docs built with Starlight. Here's what our community noticed this month:

<MonthlyShowcaseGrid starlight startDate={frontmatter.startDate} endDate={frontmatter.endDate} />

## That’s a wrap! 👋

Thanks for another great month! We can't wait to see what you build next month!

> *Want to be in this post next month? Share your Astro websites, projects, apps, tools, blog posts, demos, and videos in the `#showcase` channel in [our Discord](https://astro.build/chat) and/or tag us on social media!*

Reviewer: Fantastic job Sarah!

Just one thought: Maybe we want to inform people about sharing their site to the automatic GitHub discussion here as well immediately. Otherwise they might only share it in the Discord 🤔

But we could think about this further for the next monthly blog, no rush!👍
